packages/core/src/lib/create-mapper/get-member-path.util.ts

```diff
@@ -1,66 +1,23 @@
 import type { Selector } from '@automapper/types';
 
-/**
- * An regular expression to match will all property names of a given cleaned
- * arrow function expression. Note that if there is some computed names thus
- * they are returning if the quotes.
- *
- * @example
- * ```js
- * "s=>s.foo['bar']".match(RE_ARROW_FN_SELECTOR_PROPS)
- * // will return
- * ["foo", "'bar'"]
- * ```
- *
- * ### Explanation:
- * ```
- * (?:        // (begin of non-capturing group)
- *   (?<=     // (begin of positive lookbehind)
- *     \[     // matches a  literal "[" but without including it in the match result
- *   )        // (end of positive lookbehind)
- *   (        // (begin capturing group #1)
- *     ['"]   // match one occurrence of a single or double quotes characters
- *   )        // (end capturing group #1)
- *   (        // (begin capturing group #2)
- *     .*?    // followed by 0 or more of any character, but match as few characters as possible (which is 0)
- *   )        // (end capturing group #2)
- *   \1       // followed by the result of capture group #1
- * )          // (end of non-capturing group)
- * |          // Or matches with...
- * (?:        // (begin of non-capturing group)
- *   (?<=     // (begin of positive lookbehind)
- *     \.     // matches a literal "." but without including it in the match result
- *   )        // (end of positive lookbehind)
- *   [^.[]+   // followed by 1 or more occurrences of any character but "." nor "["
- * )          // (end of non-capturing group)
- * ```
- */
-const RE_FN_SELECTOR_PROPS = /(?:(?<=\[)(['"])(.*?)\1)|(?:(?<=\.)[^.[]+)/g;
+const PROXY_OBJECT = createProxy(() => undefined);
 
 /**
  * For a given cleaned and serialzed JS function selector expression, return a
  * list of all members that were selected.
  *
  * @returns `null` if the given `fnSelector` doesn't match with anything.
  */
-export function getMembers(fnSelectorStr: string): string[] | null {
+export function getMembers(fnSelector: Selector<any, unknown | (() => string[])>): string[] | null {
```

Comment:
nitpick: we can use `Selector<unknown, unknown | (() => string[])>` here to get rid of the `any`